dbt/include/spark/macros/materializations/table.sql

```diff
@@ -57,17 +58,26 @@ if importlib.util.find_spec("pyspark.pandas"):
   import pyspark.pandas
   pyspark_available = True
 
-# preferentially convert pandas DataFrames to pandas-on-Spark DataFrames first
+# make sure databricks.koalas exists before using it
+if importlib.util.find_spec("databricks.koalas"):
```

Comment:
The function `importlib.util.find_spec` will raise `ModuleNotFoundError` if it can't find the module.

We should use `try-except` instead?

```py
try:
    import databricks.koalas
    koalas_available = True
except ImportError:
    koalas_available = False
```

The same checks above should be the same.

Reply:
I see what you are saying @ueshin  👀 

@chamini2 is there a reason we should consider favoring `importlib.util.find_spec` over `try/catch`?

### Reproducing the report
Reading the docs for [`importlib.util.find_spec`](https://docs.python.org/3/library/importlib.html#importlib.util.find_spec), it says that:

> If name is for a submodule (contains a dot), the parent module is automatically imported.

Here's a simple reproduction case:
```python
import importlib


spec_name = "some.made.up.thing"
if importlib.util.find_spec(spec_name):
    print(f"Try to import {spec_name}")
else:
    print(f"Did not try to import {spec_name}")
```

If spec_name doesn't contain any dots (like `spec_name = "somemadeupthing"`), then it won't raise `ModuleNotFoundError`.

### Options for moving forward

Try/catch example from @ueshin:
```python
try:
    import databricks.koalas
    koalas_available = True
except ImportError:
    koalas_available = False
```

Sticking with `importlib.util.find_spec` is an eyeful:
```python
koalas_available = False
if importlib.util.find_spec("databricks") and importlib.util.find_spec("databricks.koalas"):
    import databricks.koalas
    koalas_available = True
```

Reply:
Ah, interesting!

```py
importlib.util.find_spec("databricks") and importlib.util.find_spec("databricks.koalas"):
```

works without raising exceptions!

So `pandas` (not containing any dots) and `pyspark.pandas` (`pyspark` must always be there) are safe.

Reply:
> is there a reason we should consider favoring importlib.util.find_spec over try/catch?

I don't think so! We can move to try catch for all if we want to standardize too.

The only thing is that we would be importing them, which can
1. take more time
2. add unwanted imports into the namespace

```
❯ time python -c 'import pandas'

________________________________________________________
Executed in  289.96 millis    fish           external
   usr time    1.14 secs      0.27 millis    1.14 secs
   sys time    0.76 secs      1.60 millis    0.76 secs


❯ time python -c 'import importlib; importlib.util.find_spec("pandas")'

________________________________________________________
Executed in   49.07 millis    fish           external
   usr time   34.81 millis    0.33 millis   34.48 millis
   sys time   11.32 millis    2.24 millis    9.08 mills
```

Thanks for bringing this up, @ueshin!

Reply:
Thank you for this analysis @chamini2 !

My initial reaction was, "okay, let's make it fast then!"

We need to attempt the relevant import prior to the matching `isinstance` check, so the "fast" implementation would be a bit more complicated. On the [make it work, make it right, make it fast](https://wiki.c2.com/?MakeItWorkMakeItRightMakeItFast) continuum, we'll hold off on the "make it fast" part for now.

It's good to know that we have at least 1 second of time optimization that we can achieve in cases where one or more imports are not necessary (namely when it is already a Spark DataFrame).